lib/ansible/galaxy/token.py

```diff
@@ -54,11 +59,27 @@ def __init__(self, access_token=None, auth_url=None, validate_certs=True, client
         self.access_token = access_token
         self.auth_url = auth_url
         self._token = None
+        self._issued_at = None
+        self._expires_in = None
         self.validate_certs = validate_certs
         self.client_id = client_id
         if self.client_id is None:
             self.client_id = 'cloud-services'
 
+    @property
+    def expired(self):
+        if None in (self._issued_at, self._expires_in):
+            return False
+        try:
+            issue_time = datetime.strptime(self._issued_at, ISO_8601_FORMAT_STR)
+        except ValueError:
+            display.warning(f"KeycloakToken issued at unknown time format {self._issued_at}")
+            return False
+
+        # expire a few seconds before to handle proactively
+        almost_expiration = issue_time + timedelta(seconds=self._expires_in - 3)
```

Comment:
Plz also put the numeric value in a well-readable constant.

Reply:
named `AH_TOKEN_EXPIRES_N_MIN_EARLY`, feel free to bikeshed